### PR TITLE
Added support for protected recordings

### PR DESCRIPTION
**Pull request by farhatahmad**

### CODE REVIEW

File: app/controllers/api/v1/recordings_controller.rb [hunk 1]
```diff
@@ -89,6 +89,24 @@ def recordings_count
         render_data data: count, status: :ok
       end
 
+      # GET/POST /api/v1/recordings/recording_url.json
+      def recording_url
```
Comment (KH-Amir-TN) [inline app/controllers/api/v1/recordings_controller.rb:93]: Shouldn't this API require a `ManageRecordings` permission on the given recording?

**Review by KH-Amir-TN — COMMENTED** [review]
# :fire: :fire: :fire:

File: app/controllers/api/v1/recordings_controller.rb [hunk 2]
```diff
@@ -89,6 +89,24 @@ def recordings_count
...
+        record_id = params[:record_id]
+        record_format = params[:recording_format]
+        recording = Recording.find_by(record_id:)
+
+        url = if recording.visibility == 'Protected'
+                recording = BigBlueButtonApi.new(provider: current_provider).get_recording(record_id:)
+                formats = recording[:playback][:format]
+
+                record_format.present? ? formats.find { |x| x[:type] == record_format }[:url] : formats.pluck(:url)
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/controllers/api/v1/recordings_controller.rb:102]: Yeah - I forgot to change that to format

File: app/controllers/api/v1/recordings_controller.rb (hunk 1 above)
Comment (farhatahmad) [reply app/controllers/api/v1/recordings_controller.rb:93]: Done

File: spec/controllers/recordings_controller_spec.rb [hunk 3]
```diff
@@ -268,6 +268,52 @@
...
+      end
+
+      it 'returns the formats url' do
+        recording = create(:recording, visibility: 'Published')
+        format = create(:format, recording:, recording_type: 'podcast')
+
+        post :recording_url, params: { record_id: recording.record_id, recording_format: format.recording_type }
+
+        expect(JSON.parse(response.body)['data']).to eq format.url
+      end
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply spec/controllers/recordings_controller_spec.rb:314]: It's already implied with the `allow` from above. If it wasn't called, this value wouldn't match

[the remark this answers is not in the document]
Comment (KH-Amir-TN) [reply spec/controllers/recordings_controller_spec.rb:316]: I see,
However, I still believe we may need to ensure somehow and at least that the method was called for the given recording?

If following that approach I think we might improve the test by stubbing the `get_recording` with the expected arguments (record_id).

[the remark this answers is not in the document]
Comment (farhatahmad) [reply spec/controllers/recordings_controller_spec.rb:316]: I tried stubbing the expect, but it doesn't seem to work with the `allow` at the top

[the remark this answers is not in the document]
Comment (farhatahmad) [reply spec/controllers/recordings_controller_spec.rb:316]: Either way, the value is stubbed and can only equal that value if the stubbed method is called

**Pull request merged**